akka-actor/src/main/scala/akka/io/InetAddressDnsResolver.scala

```diff
@@ -94,22 +111,66 @@ class InetAddressDnsResolver(cache: SimpleDnsCache, config: Config) extends Acto
     }
   }
 
-  override def receive = {
+  override def receive: Receive = {
+    case DnsProtocol.Resolve(_, Srv) =>
+      sender() ! Status.Failure(
+        new IllegalArgumentException(
+          "SRV request sent to InetResolver. SRV requests are only supported by async-dns resolver"))
+
+    case r @ DnsProtocol.Resolve(name, ip @ Ip(ipv4, ipv6)) =>
+      val answer = cache.cached(r) match {
+        case Some(a) => a
+        case None =>
+          try {
+            val addresses: Array[InetAddress] = InetAddress.getAllByName(name)
+            val records = addressToRecords(name, addresses.toList, ipv4, ipv6)
+            val answer = DnsProtocol.Resolved(name, records.toList)
+            if (positiveCachePolicy != Never)
+              cache.put((name, Ip()), DnsProtocol.Resolved(name, records), positiveCachePolicy)
+            answer
+          } catch {
+            case _: UnknownHostException =>
+              val answer = DnsProtocol.Resolved(name, immutable.Seq.empty)
+              if (negativeCachePolicy != Never)
+                cache.put((name, ip), answer, negativeCachePolicy)
+              answer
+          }
+      }
+      sender() ! answer
     case Dns.Resolve(name) =>
-      val answer = cache.cached(name) match {
+      // no where in akka now sends this message, but supported until Dns.Resolve/Resolved have been removed
+      val answer: Dns.Resolved = cache.cached(name) match {
         case Some(a) => a
         case None =>
           try {
-            val answer = Dns.Resolved(name, InetAddress.getAllByName(name))
-            if (positiveCachePolicy != Never) cache.put(answer, positiveCachePolicy)
+            val addresses = InetAddress.getAllByName(name)
+            // respond with the old protocol as the request was the new protocol
+            val answer = Dns.Resolved(name, addresses)
+            if (positiveCachePolicy != Never) {
+              val records = addressToRecords(name, addresses.toList)
+              cache.put((name, Ip()), DnsProtocol.Resolved(name, records), positiveCachePolicy)
+            }
             answer
           } catch {
             case _: UnknownHostException =>
               val answer = Dns.Resolved(name, immutable.Seq.empty, immutable.Seq.empty)
-              if (negativeCachePolicy != Never) cache.put(answer, negativeCachePolicy)
+              if (negativeCachePolicy != Never)
+                cache.put((name, Ip()), DnsProtocol.Resolved(name, immutable.Seq.empty), negativeCachePolicy)
               answer
           }
       }
       sender() ! answer
   }
+
+  private def addressToRecords(
+      name: String,
+      addresses: immutable.Seq[InetAddress],
+      ipv4: Boolean = true,
```

Comment:
Might there be any danger in having default values. It's a private function but always seems like bug food.

Reply:
I'll remove them